distributed/shuffle/_shuffle_extension.py

```diff
@@ -588,29 +684,111 @@ def get(
             self.columns[id] = column
             self.output_workers[id] = output_workers
             self.completed_workers[id] = set()
+            self.participating_workers[id] = output_workers.copy()
+            self._shuffle_closed_events[id] = asyncio.Event()
 
+        self.participating_workers[id].add(worker)
         return {
+            "status": "OK",
             "worker_for": self.worker_for[id],
             "column": self.columns[id],
             "schema": self.schemas[id],
             "output_workers": self.output_workers[id],
         }
 
+    async def remove_worker(self, scheduler: Scheduler, worker: str) -> None:
+        affected_shuffles = set()
+        broadcasts = []
+        participating_workers = self.participating_workers.copy()
+        for shuffle_id, shuffle_workers in participating_workers.items():
+            if worker not in shuffle_workers:
```

Comment:
Fail if any participating (input or output) worker is removed. To fail on output workers only, change this to `if worker not in self.output_workers[shuffle_id]`.

Reply:
This appears not to be tested

Reply:
Added `test_closed_bystanding_worker_during_shuffle` to test this condition.